src/hyperactive/experiment/integrations/skforecast_forecasting.py

```diff
@@ -0,0 +1,207 @@
+        Exogenous variable/s used in the evaluation experiment.
+
+    steps : int
+        Number of steps to predict.
+
+    metric : str or callable
+        Metric used to quantify the goodness of fit of the model.
+        If string, it must be a metric name allowed by skforecast
+        (e.g., 'mean_squared_error').
+        If callable, it must take (y_true, y_pred) and return a float.
+
+    initial_train_size : int
+        Number of samples in the initial training set.
+
+    refit : bool, default=False
+        Whether to re-fit the forecaster in each iteration.
+
+    fixed_train_size : bool, default=False
+        If True, the train size doesn't increase but moves by `steps` in each iteration.
+
+    gap : int, default=0
+        Number of samples to exclude from the end of each training set and the
+        start of the test set.
+
+    allow_incomplete_fold : bool, default=True
+        If True, the last fold is allowed to have fewer samples than `steps`.
+
+    return_best : bool, default=False
+        If True, the best model is returned.
+
+    n_jobs : int or 'auto', default="auto"
+        Number of jobs to run in parallel.
+
+    verbose : bool, default=False
+        Print summary figures.
+
+    show_progress : bool, default=False
+        Whether to show a progress bar.
+    """
+
+    def __init__(
+        self,
+        forecaster,
+        y,
+        steps,
+        metric,
+        initial_train_size,
+        exog=None,
+        refit=False,
+        fixed_train_size=False,
+        gap=0,
+        allow_incomplete_fold=True,
+        return_best=False,
+        n_jobs="auto",
+        verbose=False,
+        show_progress=False,
+    ):
+        self.forecaster = forecaster
+        self.y = y
+        self.steps = steps
+        self.metric = metric
+        self.initial_train_size = initial_train_size
+        self.exog = exog
+        self.refit = refit
+        self.fixed_train_size = fixed_train_size
+        self.gap = gap
+        self.allow_incomplete_fold = allow_incomplete_fold
+        self.return_best = return_best
+        self.n_jobs = n_jobs
+        self.verbose = verbose
+        self.show_progress = show_progress
+
+        super().__init__()
+
+    @classmethod
+    def get_test_params(cls, parameter_set="default"):
+        """Return testing parameter settings for the estimator.
+
+        Parameters
+        ----------
+        parameter_set : str, default="default"
+            Name of the parameter set to return.
+
+        Returns
+        -------
+        params : dict or list of dict, default = {}
+            Parameters to create testing instances of the class
+            Each dict are parameters to construct an "interesting" test instance,
+            i.e., MyClass(**params) or MyClass(**params[i]) creates a valid test
+            instance.
+            create_test_instance uses the first (or only) dictionary in `params`
+        """
+        import numpy as np
```

Comment:
this is causing test collection issues - I would use `_check_soft_dependencies` from `skbase.utils.dependencies` to fill `None` in cased `skforecast` is not present, i.e., return a `dict` that fills `None` for the required params, to avoid a failing import.